src/modules/tests/tests.effects.ts

```diff
@@ -252,15 +253,30 @@ export class TestsEffects {
             return responses.map((response, index) => {
               const matchedTests = find(completedTests, ['index', index]);
               if (response.status === HttpStatusCodes.CREATED) {
-                return new testActions.SendCompletedTestSuccess(matchedTests.slotId);
+                return matchedTests.status === TestStatus.WriteUp
+                  ? new testActions.SendPartialTestSuccess(matchedTests.slotId)
+                  : new testActions.SendCompletedTestSuccess(matchedTests.slotId);
               }
-              return new testActions.SendCompletedTestsFailure();
+              return matchedTests.status === TestStatus.WriteUp
+                ? new testActions.SendPartialTestsFailure()
+                : new testActions.SendCompletedTestsFailure();
             });
           }),
         );
     }),
   );
 
+  @Effect()
+  sendPartialTestsSuccessEffect$ = this.actions$.pipe(
```

Comment:
The effect is plural but the action is singular so should this be `sendPartialTestSuccessEffect` instead?